mongo/readconcern/readconcern.go

```diff
@@ -106,6 +113,10 @@ func New(options ...Option) *ReadConcern {
 //
 // Deprecated: Marshaling a ReadConcern to BSON will not be supported in Go Driver 2.0.
 func (rc *ReadConcern) MarshalBSONValue() (bsontype.Type, []byte, error) {
+	if rc == nil {
+		return 0, nil, ErrEmptyReadConcern
```

Comment:
Do we want to return an error here or just a BSON representation of an empty document? I see that we throw an error in the WriteConcern analogue but I'm not entirely sure on the reasoning. Wouldn't the server propagate an error if one existed because of a bad read/write concern configuration?

This is what default JSON marshaler does: https://go.dev/play/p/i9mnCDxoHk6 . 

CC: @matthewdale

Reply:
Considering the implementation of `addReadConcern`, I prefer to return an error as a fail-safe though there is a [nil check](https://github.com/mongodb/mongo-go-driver/blob/26f508fbc712a630a83bc631cf949f103d3093df/x/mongo/driver/operation.go#L1328-L1330).